### DISCUSSION TITLE
[@types/isomorphic-git__lightning-fs] LightningFS.FSConstructorOptions is overly restrictive (and possibly others)

### DISCUSSION THREAD

**Discussion by jesseditson** [DefinitelyTyped/DefinitelyTyped#58917] [Issues with a @types package]
Using `@types/isomorphic-git__lightning-fs` in a project today and noticed that the type for `LightningFS.FSConstructorOptions` is incorrect in that it implicitly defines all properties as required, when the actual lightning-fs package has defaults for all values.

The [source repo](https://github.com/isomorphic-git/lightning-fs) appears to have a typedef on the main branch, but has not been published to npm yet, so the @types package is the current best option.

Here's the @types def:

https://github.com/DefinitelyTyped/DefinitelyTyped/blob/136ec91b7aadd50a5acfc60a5d07272a647d44cd/types/isomorphic-git__lightning-fs/index.d.ts#L47-L69

Which just generally needs `?` appended to all the keys.

I'm not familiar enough with the lib to make the change yet, and it's possible that the actual solution is to just publish the first-party types, but I didn't see a related issue so I figured I'd open this one for my own tracking purposes.

[a comment by an automated account is not shown]

**Comment by jesseditson** [comment]
(also posted a comment on the relevant upstream repo, to attempt to tie this stuff together: https://github.com/isomorphic-git/lightning-fs/issues/48)

**Reply by 0xTheProDev** [reply]
I have raised a PR here: https://github.com/DefinitelyTyped/DefinitelyTyped/pull/59295
